.editorconfig

```diff
@@ -50,17 +53,19 @@ align_array_table = false
 ## Other indentation settings
 never_indent_before_if_condition = false
 never_indent_comment_on_if_branch = false
+keep_indents_on_empty_lines = false
+allow_non_indented_comments = true
 
 ## Line spacings
-line_space_after_if_statement = max(1)
-line_space_after_do_statement = max(1)
-line_space_after_while_statement = max(1)
-line_space_after_repeat_statement = max(1)
-line_space_after_for_statement = max(1)
-line_space_after_local_or_assign_statement = max(1)
-line_space_after_function_statement = fixed(1)
-line_space_after_expression_statement = max(1)
-line_space_after_comment = max(1)
+line_space_after_if_statement = max(2)
+line_space_after_do_statement = max(2)
+line_space_after_while_statement = max(2)
+line_space_after_repeat_statement = max(2)
+line_space_after_for_statement = max(2)
+line_space_after_local_or_assign_statement = max(2)
+line_space_after_function_statement = max(2)
+line_space_after_expression_statement = keep
+line_space_after_comment = keep
```

Comment:
Several of these don't make sense, but particularly the function spacing.

Reply:
Since code formatting is a matter of opinion, I don't think it's a question of whether it makes sense but rather whether it improves readability.

In this case, I was saving files under the original editorconfig and it was stripping out so much whitespace that all the code was running together as a single block, which for me makes it way harder to read.

I simply chose the less opinionated option here, so if someone really wants to format code in close blocks by hand they can, but at least the autoformatter won't *force* it on you.

Reply:
Yeah the original values using `1` broke because the EmmyLua behaviour is unfortunately not intuitive, but is sadly considered not-a-bug (see my edit): https://github.com/CppCXY/EmmyLuaCodeStyle/issues/191#issuecomment-2516038729

So most of it in the updated version is actually fine, but the `keep` values don't really make sense since that allows an infinite amount of empty lines to be added in some cases when it should be stripped down to 1, so everything should probably have `max(2)`.